File: 프로그래머스/2019 KAKAO BLIND RECRUITMENT/오픈채팅방.py
# 오픈채팅방
from collections import defaultdict
def solution(record):
    answer = []
    dict = defaultdict(list)
    for r in record:
        r = r.split(" ")
        if r[0] != 'Leave':
            dict[r[1]] = r[2]
    for r in record:
        r = r.split(" ")
        if r[0] == 'Change':
            continue
        if r[0] == 'Enter':
            answer.append(dict[r[1]]+'님이 들어왔습니다.')
        else:
            answer.append(dict[r[1]]+'님이 나갔습니다.')
    return answer

File: 프로그래머스/2019 KAKAO BLIND RECRUITMENT/오픈채팅방.js
// 오픈채팅방
function solution(record) {
    let answer = [];
    let chat = new Map();
    record.forEach((ele) => {
        ele = ele.split(' ');
        if(ele[0] != 'Leave') chat.set(ele[1],ele[2]);
    });
    record.forEach((ele) => {
        ele = ele.split(' ');
        if(ele[0] === 'Enter') answer.push(`${chat.get(ele[1])}님이 들어왔습니다.`);
        if(ele[0] === 'Leave') answer.push(`${chat.get(ele[1])}님이 나갔습니다.`);
    });
    return answer;
}